fs/sync/sync.go

```diff
@@ -399,6 +402,11 @@ func (s *syncCopyMove) pairChecker(in *pipe, out *pipe, fraction int, wg *sync.W
 					fs.Errorf(pair.Dst, "Source and destination exist but do not match: %v", err)
 					s.processError(err)
 				} else {
+					if pair.Dst != nil {
+						s.markDirModified(path.Dir(filepath.ToSlash(pair.Dst.Remote())))
```

Comment:
This should not need `filepath.ToSlash` here - all paths should be unix `/` delimited in rclone internals.

If they aren't then that is a bug somewhere else :-)

Please remove them all from this file! Also see later suggestion for a function `markDirModifiedObject`...

Reply:
I removed all the `filepath` usages here :) I think the error I was seeing the other day may just have been a transient fluke of GitHub Actions (I've been seeing more of those lately)

Here is an example of what I mean though about the `path` library causing issues on Windows with rclone's unix paths: [this test](https://github.com/nielash/rclone/commit/c575274b07c391861b154347a9d5382f9fe84204) fails [only on Windows](https://github.com/nielash/rclone/actions/runs/8153440329). (And I confirmed it also fails on a real PC, not just GH Actions)

I think the issue in this particular case is `path.Join` removing the double `//` and then rclone adding it back incorrectly. But there may be other issues as well.

I think it would be really handy to have a more comprehensive library of rclone-safe path manipulation functions!